staging/src/k8s.io/apiserver/pkg/endpoints/groupversion.go

```diff
@@ -135,11 +136,18 @@ func removeNonPersistedResources(infos []*storageversion.ResourceInfo) []*storag
 
 // staticLister implements the APIResourceLister interface
 type staticLister struct {
-	list []metav1.APIResource
+	list      []metav1.APIResource
+	hash      string
+	hashToken sync.Once
```

Comment:
rather than trying to synchronize here, and assuming this is the path we want to move forward with, you could create a "new" method to hash and set the `list` and `hash`. That wouldn't be lazy, not sure how bad that is.

Reply:
Also, we'll almost certainly need the hash before someone hits the endpoint (to populate the ServerGroups), so I don't know if lazy is useful here.

Reply:
hash is now eagerly evaluated. marking as resolved